utils/http.go

```diff
@@ -66,12 +65,13 @@ func (root *rootHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if authCtx, err = root.auth.Authorized(ctx, access...); err != nil {
 			if challenge, ok := err.(auth.Challenge); ok {
 				// Let the challenge write the response.
-				challenge.ServeHTTP(w, r)
+				challenge.SetHeaders(w)
 
-				w.WriteHeader(http.StatusUnauthorized)
+				if err := errcode.ServeJSON(w, errcode.ErrorCodeUnauthorized.WithDetail(access)); err != nil {
```

Comment:
If we aren't going to do anything if the error isn't nil, should we just call it without assigning the `err` variable?